### PR TITLE
RSDK-2027 - Set up cartographer integration tests automation in CI

### PR DESCRIPTION
**Pull request by kkufieta**
Ticket: https://viam.atlassian.net/browse/RSDK-2027

NOTE: This PR has some of the changes in it that are already done in this PR: https://github.com/viamrobotics/viam-cartographer/pull/14. This will get fixed once PR 14 is merged. I will point out the actual changes in the code - you can ignore the rest.

Done:
* Set up cartographer integration tests automation in CI
    * I am keeping running the same integration tests against the code in RDK until we are ready to delete builtin from RDK.

PR for testing the CI: https://github.com/viamrobotics/viam-cartographer/pull/17

### CODE REVIEW

File: .artifact/config.json
Comment (kkufieta) [inline .artifact/config.json:1]: Ignore

File: .artifact/tree.json
Comment (kkufieta) [inline .artifact/tree.json:1]: Ignore

File: .github/workflows/pullrequest.yml
Comment (kkufieta) [inline .github/workflows/pullrequest.yml:1]: Will revert back to `main` once PR is approved

File: .github/workflows/test.yml
Comment (kkufieta) [inline .github/workflows/test.yml:1]: This is the only actual change in this PR!

File: viam-cartographer.go
Comment (kkufieta) [inline viam-cartographer.go:1]: Ignore

File: viam-cartographer_test.go
Comment (kkufieta) [inline viam-cartographer_test.go:1]: Ignore

**Review by kkufieta — COMMENTED** [review]
Some notes on what to focus on

**Comment by zaporter-work** [comment]
The RDK has a special rule to ignore `staticcheck` in `rdk/services/slam`. This rule was not brought over to our viam-cartographer and viam-orb-slam3 repos. Do you think it is better to throw `nolint:staticcheck` around or edit the `golangci.yaml`?

```
    # TODO(RSDK-1053): reenable the staticcheck linter for this directory
    # once deprecated API is no longer used.
    - path: services/slam/
      linters:
        - staticcheck
```

**Review by zaporter-work — APPROVED** [review]

**Comment by kkufieta** [comment]
@zaporter-work 
> The RDK has a special rule to ignore `staticcheck` in `rdk/services/slam`. This rule was not brought over to our viam-cartographer and viam-orb-slam3 repos. Do you think it is better to throw `nolint:staticcheck` around or edit the `golangci.yaml`?
> 
> ```
>     # TODO(RSDK-1053): reenable the staticcheck linter for this directory
>     # once deprecated API is no longer used.
>     - path: services/slam/
>       linters:
>         - staticcheck
> ```

Great question - @EshaMaharishi asked the same one here: https://github.com/viamrobotics/viam-cartographer/pull/14#discussion_r1137213609

I believe that it is sufficient to disable `staticcheck` for those few local lines rather than disabling it globally. These lines will be removed with the ticket I created (linked in Esha's comment).

**Comment by kkufieta** [comment]
Closing this PR in favor of the following, cleaner PR: https://github.com/viamrobotics/viam-cartographer/pull/20

**Pull request closed without merging**